x/ecocredit/server/marketplace/update_sell_orders_test.go

```diff
@@ -0,0 +1,262 @@
+	assert.NilError(t, err)
+
+	// cannot edit the sell order with this address
+	_, err = s.k.UpdateSellOrders(s.ctx, &marketplace.MsgUpdateSellOrders{
+		Owner: unauthorized.String(),
+		Updates: []*marketplace.MsgUpdateSellOrders_Update{
+			{SellOrderId: 1, NewQuantity: "1"},
+		},
+	})
+	assert.ErrorContains(t, err, sdkerrors.ErrUnauthorized.Error())
+}
+
+func TestUpdateSellOrder_AskPrice(t *testing.T) {
+	t.Parallel()
+	s := setupBase(t)
+	testSellSetup(t, s, batchDenom, ask.Denom, ask.Denom[1:], classId, start, end, creditType)
+
+	s.paramsKeeper.EXPECT().GetParamSet(any, any).Do(func(any interface{}, p *ecocredit.Params) {
+		p.CreditTypes = []*ecocredit.CreditType{&creditType}
+	}).Times(2)
+	expiration := time.Now()
+	_, err := s.k.Sell(s.ctx, &marketplace.MsgSell{
+		Owner: s.addr.String(),
+		Orders: []*marketplace.MsgSell_Order{
+			{BatchDenom: batchDenom, Quantity: "5.22", AskPrice: &ask, DisableAutoRetire: false, Expiration: &expiration},
+		},
+	})
+	assert.NilError(t, err)
+
+	orderBefore, err := s.marketStore.SellOrderTable().Get(s.ctx, 1)
+	// can update price of same denom
+	askUpdate := sdk.NewInt64Coin(ask.Denom, 25)
+	_, err = s.k.UpdateSellOrders(s.ctx, &marketplace.MsgUpdateSellOrders{
+		Owner: s.addr.String(),
+		Updates: []*marketplace.MsgUpdateSellOrders_Update{
+			{SellOrderId: 1, NewAskPrice: &askUpdate},
+		},
+	})
+	assert.NilError(t, err)
+	orderAfter, err := s.marketStore.SellOrderTable().Get(s.ctx, 1)
+	assert.NilError(t, err)
+	assert.Equal(t, orderAfter.AskPrice, askUpdate.Amount.String())
+	assert.Equal(t, orderBefore.MarketId, orderAfter.MarketId)
+
+	orderBefore, err = s.marketStore.SellOrderTable().Get(s.ctx, 1)
+	assert.NilError(t, err)
+
+	askUpdate = sdk.NewInt64Coin("ubar", 18)
+	_, err = s.k.UpdateSellOrders(s.ctx, &marketplace.MsgUpdateSellOrders{
+		Owner: s.addr.String(),
+		Updates: []*marketplace.MsgUpdateSellOrders_Update{
+			{SellOrderId: 1, NewAskPrice: &askUpdate},
+		},
+	})
+	assert.NilError(t, err)
+
+	order, err := s.marketStore.SellOrderTable().Get(s.ctx, 1)
+	assert.NilError(t, err)
+
+	assert.Equal(t, order.AskPrice, askUpdate.Amount.String())
+	assert.Equal(t, order.MarketId, orderBefore.MarketId+1)
+}
+
+func TestUpdateSellOrder_Expiration(t *testing.T) {
+	t.Parallel()
+	s := setupBase(t)
+	testSellSetup(t, s, batchDenom, ask.Denom, ask.Denom[1:], classId, start, end, creditType)
+
+	s.paramsKeeper.EXPECT().GetParamSet(any, any).Do(func(any interface{}, p *ecocredit.Params) {
+		p.CreditTypes = []*ecocredit.CreditType{&creditType}
+	}).Times(1)
+
+	future := time.Date(2077, 1, 1, 1, 1, 1, 1, time.Local)
+	middle := time.Date(2022, 1, 1, 1, 1, 1, 1, time.Local)
+	past := time.Date(1970, 1, 1, 1, 1, 1, 1, time.Local)
+
+	_, err := s.k.Sell(s.ctx, &marketplace.MsgSell{
+		Owner: s.addr.String(),
+		Orders: []*marketplace.MsgSell_Order{
+			{BatchDenom: batchDenom, Quantity: "5.22", AskPrice: &ask, DisableAutoRetire: false, Expiration: &past},
+		},
+	})
+	assert.NilError(t, err)
```

Comment:
Creating a sell order with an expiration in the past? This should fail...?

Reply:
I havent set the blocktime at this point so it'll compare against the default 0 time value, i can add a comment to clear that up